Added realtime_seo, removed layout override capability, added update …

mechanoid23 opened the pull request:
…hook.

mechanoid23 committed:
Added realtime_seo, removed layout override capability, added update hook.

mechanoid23 committed:
Naming inconsistencies, machine name.

mechanoid23 committed:
Config file name correction.

mechanoid23 committed:
Several.

emerham commented on osu_story.install:
```diff
@@ -40,3 +40,26 @@ function osu_story_update_9002(&$sandbox) {
+  /** @var \Drupal\Core\Config\CachedStorage $config_storage */
+  $config_storage = \Drupal::service('config.storage');
+  $config_storage->write('core.entity_form_display.node.story.default', $source->read('core.entity_form_display.node.story.default'));
+  $config_storage->write('core.entity_view_display.node.story.default', $source->read('core.entity_view_display.node.story.default'));
```
I want to remove the overrides but we cannot just update the form for all existing sites this way. Firstly we'd have to "roll-back" all overrides which is too authoritative for us to do right now. So lets just remove this one line.

emerham commented on config/install/views.view.osu_stories.yml:
```diff
@@ -11,6 +11,7 @@ dependencies:
     - taxonomy.vocabulary.tags
   module:
     - date_ap_style
+    - group
```
I understand why we are adding group as a required dependency, but this also makes it so group is a required install, which is currently not required. What we can do is move the Group landing page view into an optional view that has a dependency on group.

emerham committed:
Add layout override checks and optional configuration for OSU Story module

- Introduced logic to conditionally write default entity view display configuration based on layout overrides.
- Added handling for optional configuration, including new `views.view.osu_stories__groups`.
- Removed unused dependencies and parameters from views configuration.
- Updated core version requirement to `^10 || ^11`.

emerham reviewed (approved):
I made changes to ensure that we will not be destructive in our update.
